### PR TITLE
Headers.php - fromString() method bug

### PR DESCRIPTION
**Pull request by weierophinney**
Broke on May 7th in https://github.com/zendframework/zend-mail/commit/f451a87ac99db74a53dfa9a01451c01cca004fb7

The new fromString() method in Headers.php will throw a runtime exception if $string contains more than 1 new line whether they are consecutive or not.

Postfix NDR email that breaks it:
```
Date: Wed, 21 Aug 2013 00:22:27 -0700
From: Mail Delivery Subsystem <MAILER-DAEMON@localhost.localdomain>
Message-Id: <201308210722.r7L7Lpmx022081@localhost.localdomain>
To: <root@localhost.localdomain>
MIME-Version: 1.0
Content-Type: multipart/report; report-type=delivery-status;boundary="r7L7Lpmx022081.1377069747/localhost.localdomain"
Subject: Returned mail: see transcript for details
Auto-Submitted: auto-generated (failure)

This is a MIME-encapsulated message

--r7L7Lpmx022081.1377069747/localhost.localdomain

The original message was received at Fri, 16 Aug 2013 00:00:48 -0700
from localhost.localdomain [127.0.0.1]

   ----- The following addresses had permanent fatal errors -----
<sysadmin@example.com>

   ----- Transcript of session follows -----
<sysadmin@example.com>... Deferred
Message could not be delivered for 5 days
Message will be deleted from queue

--r7L7Lpmx022081.1377069747/localhost.localdomain
Content-Type: message/delivery-status

Reporting-MTA: dns; localhost.localdomain
Arrival-Date: Fri, 16 Aug 2013 00:00:48 -0700

Final-Recipient: RFC822; sysadmin@internet.com
Action: failed
Status: 4.4.7
Last-Attempt-Date: Wed, 21 Aug 2013 00:22:27 -0700

--r7L7Lpmx022081.1377069747/localhost.localdomain
Content-Type: message/rfc822

Date: Fri, 16 Aug 2013 00:00:48 -0700
From: root <root@localhost.localdomain>
Message-Id: <201308160700.r7G70msc008297@localhost.localdomain>
To: sysadmin@example.com
Subject: MySQL Backup Log for example.com - 2013-08-16_00h00m

======================================================================
AutoMySQLBackup VER 2.5
http://sourceforge.net/projects/automysqlbackup/

Backup of Database Server - test.com
======================================================================
Backup Start Time Fri Aug 16 00:00:01 PDT 2013
======================================================================
Daily Backup of Database ( information_schema )
Rotating last weeks Backup...
removed `/backups'

----------------------------------------------------------------------
Daily Backup of Database ( confluence )
Rotating last weeks Backup...
removed `/backupnce.bz2'

----------------------------------------------------------------------
Daily Backup of Database ( crucible )
Rotating last weeks Backup...
removed `/backups/daily/.bz2'

----------------------------------------------------------------------
Daily Backup of Database ( jira )
Rotating last weeks Backup...
removed `/bal.bz2'

----------------------------------------------------------------------
Daily Backup of Database ( jira_4_2 )
Rotating last weeks Backup...
removed `/backups/dail.sql.bz2'

----------------------------------------------------------------------
Daily Backup of Database ( mysql )
Rotating last weeks Backup...
removed `/backups/dql.bz2'

----------------------------------------------------------------------
Backup End Fri Aug 16 00:00:48 PDT 2013
======================================================================
Total disk space used for backup storage..
Size - Location
62.21T /backups

======================================================================
If you find AutoMySQLBackup valuable please make a donation at
http://sourceforge.net/project/project_donations.php?group_id=101066
======================================================================

--r7L7Lpmx022081.1377069747/localhost.localdomain--
```

### CODE REVIEW

**Comment by weierophinney** [comment]
I've attached a patch to this issue (converting the issue into a PR), using the email you provided in the description, passing it to `Message::fromString()`, and performing assertions on the headers. Currently, the tests pass.

It's possible my test case is not testing your specific scenario, however, @LassoMike , so please review and verify, or indicate how I should change the tests.

**Comment by weierophinney** [comment]
@LassoMike ping.

**Comment by LassoMike** [comment]
It looks good, but I am going to test some more on my end. It is curious that your test passes, but mine did not!

**Comment by Maks3w** [comment]
Sometimes text editor may strip some whitespace or the message retrieved from one protocol is slight different compared with web source.

File: test/MessageTest.php [hunk 1]
```diff
@@ -790,4 +790,29 @@ public function testHeaderUnfoldingWorksAsExpectedForMultipartMessages()
...
         $this->assertContains($multipartContent->getMime()->boundary(), $contentType->getFieldValue());
     }
+
+    /**
+     * @group 19
+     */
+    public function testCanParseMultipartReport()
+    {
+        $raw = file_get_contents(__DIR__ . '/_files/zend-mail-19.txt');
+        $message = Message::fromString($raw);
```
Comment (LassoMike) [inline test/MessageTest.php]: The exception is being thrown by Headers::fromString(), not Message::fromString().

**Comment by LassoMike** [comment]
This error is being triggered when:
```
$raw = file_get_contents(__DIR__ . '/_files/zend-mail-19.txt');
$part = new Part(['raw' => $raw]);
$part->countParts();
```

**Comment by LassoMike** [comment]
@weierophinney ping. Can I help in any other way? Was that enough info?

**Comment by weierophinney** [comment]
@LassoMike I'll give that a try, but may be a while (long TODO list currently).

**Comment by mikesul** [comment]
I've run into the same issue. I think the bug is actually in Zend\Mime\Decode::splitMessage as it has handling to try to detect if the first line is a header, but it skips over the first line if the message starts with a line break (due to how strtok works).

As such, I've submitted a PR to zendframework/zend-mime#7 with a workaround.

**Comment by weierophinney** [comment]
I've merged the change, mainly because I think it's a good test case for us to have. Unfortunately, it works on all supported versions of zend-mime, so that leaves me with a "cannot reproduce" status.

If either @mikesul or @LassoMike can come up with one, please create a PR with the failing test.

Thanks!

**Pull request merged**